### ISSUE TITLE
rbenv init creates duplicates in PATH

### ISSUE THREAD

**Issue by RichardFevrier** [rbenv/rbenv#1573]
Hello, adding `rbenv init - | source` in `config.fish` adds twice the `shims` path (same issue on `nodenv` but not on `pyenv`).

**Comment by mislav** [comment]
Hi, if you run `rbenv init - fish`, you can inspect the script that will be eval'd by your shell. This is it for me:

```fish
set -gx PATH '/Users/mislav/.rbenv/shims' $PATH
set -gx RBENV_SHELL fish
command rbenv rehash 2>/dev/null
function rbenv
  set command $argv[1]
  set -e argv[1]

  switch "$command"
  case rehash shell
    rbenv "sh-$command" $argv|source
  case '*'
    command rbenv "$command" $argv
  end
end
```

I only see the shims path being added once. Could it be that something else is adding another shims path, or that you're running a shell within a shell and that both edited PATH?

**Comment by jasonkarns** [comment]
It sounds like rbenv init is being invoked twice.

`rbenv-init` should not "create duplicates" but rather proceeds to prepend the sims to PATH regardless if they're already present. If your shell setup is already calling rbenv-init elsewhere, then the second invocation would create a second entry in PATH.

When `rbenv-init` runs, its job is to place the shims at the beginning of `PATH`. If it's called twice, they'll be added twice. The recommended fix is to not call rbenv-init twice.

Some more background: rbenv initially tried to be "clever" and only add the shims if they weren't present: https://github.com/rbenv/rbenv/commit/03fa148e814d50aba8012dc94b866213a00ea8a9

This causes no end of problems thanks to poor handling of *rc and *profile files by some OS configurations (most notably, OS X back in the day). https://github.com/rbenv/rbenv/issues/369
See also the general problem: https://stackoverflow.com/questions/13058578/how-to-prevent-tmux-from-filling-up-the-global-path-variable-with-duplicated-pat

So the de-duping was subsequently reverted: https://github.com/rbenv/rbenv/commit/e2173df4aa91c8d365ca1596fb857fcac9fdd787

Current behavior is now intentional (`rbenv-init` simply prepends to `PATH` regardless of existing value).

**Comment by ipatch** [comment]
so i kinda ran into this myself this afternoon with my setup. a couple of things that helped me

1. setting the fish_trace var to 2 to debug my fish startup sequence https://github.com/fish-shell/fish-shell/issues/3427#issuecomment-520176772
2. i manage all my path entries manually and do not use fish shells universal variables, so when i set a path entry in fish i do something like the below,

```
set -gx PATH /some/path/bin $PATH
```

in my config.fish i create a local var with all the paths i'd like to have in my interactive shell, and if the dir exists their added, but the dir does not exists their not added, which works great for various setups. because on some boxes i use rbenv and others i use rvm.

3. another thing that came is the `fish_user_paths`

that's something that can contain path entries as well, so check that `echo $fish_user_paths`

basically i want all my path entries stored in the `PATH` var and nothing else. that said for my use case with my setup, i just added the below two paths to my config.fish

```
$HOME/.rbenv/bin
$HOME/.rbenv/shims
```

i did the git clone install for this particular box (asahi linux m1)

and every time i now reload my shell using `exec $SHELL` or `exec fish` the path entry of `$HOME/.rbenv/shims` is not being duplicated.

however if i go with the setup described in the readme and add any of the below lines within my `config.fish` i'll get duplicate entries in my path env var ie. `PATH`

```
rbenv init - | source
eval "$(rbenv init -)"
rbenv init -
```

i think the easiest way to prevent getting  bit by duplicate path entries is to manage this yourself 🤷‍♂️
